### PR TITLE
[Reviewer: Matt] Hide netsnmp behind interfaces

### PR DESCRIPTION
**Pull request by rkd-msw**
I've tested in clearwater-fv-test, by forcing a compile-time error if READ is defined after including the client-code-facing header files that should be interface-only.

### CODE REVIEW

File: include/snmp_table.h [hunk 1]
```diff
@@ -38,6 +38,7 @@
 #include <map>
 #include <string>
 
+#include "snmp_row.h"
 #include "snmp_includes.h"
```
Comment (mirw) [inline include/snmp_table.h]: Worth a comment in the header that this is not intended to be included by core code and why?

Comment (rkd-msw) [inline include/snmp_table.h]: I wonder if it's actually worth shoving all these headers into a `include/snmp_internals` subfolder.

Comment (mirw) [inline include/snmp_table.h]: Yes, that sounds good.  Although it's still worth commenting why.

**Comment by mirw** [comment]
Looks good.  I eye-balled the code moves, but didn't check them line-by-line - are you confident about those moves?  (I guess so, but wanted to check.)

**Comment by rkd-msw** [comment]
Yes - I've confirmed that the clearwater-fv-test tests still compile (with the change to call create()), and that READ isn't defined after including the 'public' headers.

**Pull request merged**